apollo-router/src/services/subgraph_service.rs

```diff
@@ -1014,9 +1017,10 @@ pub(crate) async fn process_batch(
                 .map(|mut http_res| {
                     *http_res.headers_mut() = parts.headers.clone();
                     // Use the original context for the request to create the response
-                    let context = contexts.pop().expect("we have a context for each response");
+                    let (context, id) =
+                        contexts.pop().expect("we have a context for each response");
```

Comment:
needs a test around batching

Reply:
batching test added in [98a4ba3](https://github.com/apollographql/router/pull/5858/commits/98a4ba3004a31808a18df7d55d2a46431d5473e8)